src/models/unified_api_server.py

```
@@ -0,0 +1,443 @@
+        ):
+            """Transcribe audio using Whisper model."""
+            if not self.models["transcriber"]:
+                raise HTTPException(status_code=503, detail="Transcription model not available")
+
+            # Validate file type
+            if not file.filename.lower().endswith(('.mp3', '.wav', '.m4a', '.ogg', '.flac')):
+                raise HTTPException(status_code=400, detail="Unsupported audio format")
+
+            try:
+                # Save uploaded file temporarily
+                temp_path = f"/tmp/{file.filename}"
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/"

(S108)